core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java

```diff
@@ -349,12 +348,8 @@ private static Map<SchemaTableName, List<ColumnMetadata>> doListTableColumns(Ses
 
     private static TrinoException handleListingException(RuntimeException exception, String type, String catalogName)
     {
-        ErrorCodeSupplier result = GENERIC_INTERNAL_ERROR;
-        if (exception instanceof TrinoException trinoException) {
-            result = trinoException::getErrorCode;
-        }
         return new TrinoException(
-                result,
+                findErrorCode(exception),
```

Comment:
Add in the description of this PR the stacktrace which motivated you to make this change.